internals/servicelog/ringbuffer.go

```diff
@@ -73,6 +75,10 @@ func (rb *RingBuffer) Close() error {
 func (rb *RingBuffer) Closed() bool {
 	rb.rwlock.RLock()
 	defer rb.rwlock.RUnlock()
+	return rb.closed()
+}
+
+func (rb *RingBuffer) closed() bool {
```

Comment:
Given that it's a simple attribute access, there's arguably no need to protect it with a lock

Reply:
I don't think that's true in general. It might work here, as it's a single-machine-word `bool` type, but it doesn't seem like good practice to rely on this. For example, a string, slice, or interface attribute would be a "simple attribute", but 2 or 3 machine words.